### PR TITLE
Benchmark script: use process groups to manage children

### PR DESCRIPTION
**Pull request by smowton**
Previously when a process timed out only the managing shell would be killed;
the underlying process would continue, starving later jobs of CPU time

### CODE REVIEW

**Review by allredj — COMMENTED** [review]
:heavy_check_mark:
Passed Diffblue compatibility checks (cbmc commit: 0a6286780343a21c062aa030b7bd89154a797e71).
Build URL: https://travis-ci.com/diffblue/test-gen/builds/102491205

**Comment by tautschnig** [comment]
Could we instead step-by-step Python-ize this code? Just completely doing the timed execution in Python:
````python
import subprocess
import threading
import time

def run_cmd_with_timeout(cmd, timeout_sec):
    """
    Execute `cmd` in a subprocess and enforce timeout `timeout_sec` seconds.
    Return subprocess exit code on natural completion of the subprocess.
    Raise an exception if timeout expires before subprocess completes.
    """
    t1 = time.time()
    proc = subprocess.Popen(cmd)
    timer = threading.Timer(timeout_sec, proc.kill)
    try:
        timer.start()
        proc.communicate()
    finally:
        timer.cancel()
    t2 = time.time()
    return (proc.returncode, t2 - t1)
````

**Comment by smowton** [comment]
Fine with me -- if you want to make a PR to replace this whole script I'd vote for that one over this one!

**Comment by smowton** [comment]
If I can get a code owner review I'll merge this for now pending a rewrite

**Review by thk123 — APPROVED** [review]

**Pull request merged**
